src/test/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs

```diff
@@ -206,6 +238,22 @@ private string PrepareAppHostMockFile(TestDirectory testDirectory, Action<byte[]
             0, 112, 2, 0, 0, 4, 0, 0, 0, 0, 0, 0, 3, 0, 96, 193, 0, 0, 24,
             0, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0 };
 
+        private static int GetFilePermissionOctal(string path)
+        {
+            var fileInfo = new FileInfo(path);
+            var _ = fileInfo.IsReadOnly; // this is to implicitly initialize FileInfo -> FileSystem -> fielStatus instance
+
+            s_fileSystem_fileStatusField = typeof(FileSystemInfo).GetField("_fileStatus", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
We should not use private reflection - if the implementation details in CoreFX change the test will break - in weird ways.

If there's no public API in CoreFX to do what we want, then the better solution is to call the native API through PInvoke directly here.

Reply:
Hmm - I missed your comment before I posted by feedback.
I still don't like the private reflection (It's not about perf - at all). We would basically need a copy of https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/Interop/Unix/System.Native/Interop.Stat.cs to do this through the PInvoke. I think that's acceptable - especially if there's a chance there will be a blessed way to do it one day based on  dotnet/corefx#26394.

Reply:
> call the native API through PInvoke directly here.

The `stat struct` makes it hard to do a direct pinvoke.

We can call the system `stat` program. You can find some code here: https://github.com/NuGet/NuGet.Client/blame/3803820961f4d61c06d07b179dab1d0439ec0d91/test/NuGet.Core.Tests/NuGet.Common.Test/DirectoryUtilityTests.cs#L55-L87.

Reply:
Sorry - I missed the fact the CoreFX does this through a native helper... oh my.
OK - as much as I don't like private reflection... I guess I'll live with it for this one...
If you keep the private reflection, please wrap it in a try/catch and throw an exception which will explain that this relies on private reflection to CoreFX and why - and maybe add a more detailed comment above it. I want to make sure that if this does break, it's relatively easy to figure out why.

Reply:
Refactored this logic in a private nested class with try/catch and better error handling.